tests/bsim/bluetooth/host/central/src/main.c

```diff
@@ -19,6 +24,19 @@ static void connected_cb(struct bt_conn *conn, uint8_t err)
 	bt_conn_unref(conn);
 }
 
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	TEST_ASSERT(conn);
+	TEST_ASSERT(err == 0, "Expected connection establishment");
```

Comment:
It's a bit interesting that you do `TEST_ASSERT(conn);` instead of `TEST_ASSERT(conn != NULL);` and/or `TEST_ASSERT(err == 0` instead of `TEST_ASSERT(!err` :)

To be more explicit you could also do 
```suggestion
	TEST_ASSERT(conn);
	TEST_ASSERT(err == BT_HCI_ERR_SUCCESS, "Expected connection establishment");
```